lib/puppet-lint/plugins/check_parameter_documentation.rb

```diff
@@ -3,15 +3,20 @@ def check
     class_indexes.concat(defined_type_indexes).each do |idx|
       next if idx[:param_tokens].nil?
 
-      doc_params = []
+      doc_params = {}
+      doc_params_duplicates = []
       is_private = false
-      tokens[0..idx[:start]].reverse_each do |dtok|
+      tokens[0..idx[:start]].each do |dtok|
```

Comment:
I can't figure out why it was parsed in reverse order before??

Reply:
It was introduced in 8009750386fc1a1b9ab53331d5d42e1c6c45a979 (initial commit). I suspect that most tools read the last docs first and it was based on that, but I don't really see a good reason.

Reply:
I've backed this out now that I'm warning on *all* instances of the duplicate.